### DISCUSSION TITLE
ContextActionsProvider auto suggestion value not working

### DISCUSSION THREAD

**Discussion by nwoik** [eclipse-glsp/glsp#1344] [General]
When I right click and enter text, the auto suggestions and auto complete works. But when I press enter, it does nothing. I'm unable to enter the value the ContextActionsProvider provides. If I manually type without the provider, I can input my values just fine. The issue is  just any auto completed values.

Am I missing something extra here. Why am I not able to enter my selected values?

```java
public class CustomContextActionProvider implements ContextActionsProvider {


   @Override
   public String getContextId() { return "editor"; }

   @Inject
   protected EMFNotationModelState modelState;

   @Override
   public List<? extends LabeledAction> getActions(final EditorContext editorContext) {

      ArrayList<SetAutoCompleteValueAction> autoCompletions = new ArrayList<>();


      for (int i = 0; i < 10; i++) {
         autoCompletions.add(new SetAutoCompleteValueAction("" + i, "", ""+ i));
      }

      return autoCompletions;
   }

}
```

**Comment by tortmayr** [comment]
Hi @nwoik 

at first glance your implementation looks correct and you don't seem to miss anything.
Cloud you share additional details about your project so that we can try to reproduce the issue?
- glsp version
- integration scenario? Java-emf with what frontend? Theia/VS Code/Standalone

Thanks!

**Reply by nwoik** [reply]
Hi @tortmayr

Sorry for the late response. I'm using version 2.1 for both GLSP and my Theia integration.
The project is heavily based off the example [java-emf-theia](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia)

Thanks

**Reply by tortmayr** [reply]
Thanks, I will try to reproduce/isolate the issue and come back to you.

**Reply by tortmayr** [reply]
I looked into this and it seems that the version bump  of autocompleter to 9.x causes the issue (https://github.com/kraaden/autocomplete/blob/master/changelog.md#breaking-changes).

At the moment there is not quickfix/workaround availble and we have to further look into this and update the `AutocompleteWidget` to conform to the breaking changes.

Could you quickly share the UI extension code that you are using on the frontend to render the suggestions provided by your `CustomContextActionsProvider`? I would like to double check to ensure that the problem your are encountering is caused by the same issue.

**Reply by nwoik** [reply]
I'm pretty much using the exact same UI extension frontend setup as the [Coffee Editor](https://github.com/eclipse-emfcloud/coffee-editor/blob/master/client/coffee-workflow-glsp/src/direct-task-editing/direct-task-editor.ts)
